### ISSUE TITLE
Few Questions

### ISSUE THREAD

**Issue by aljawaid** [bilde2910/Hauk#103]
Hello, great concept here, but I have a few questions....

* If the install is done manually on a vps/cpanel account, can it be installed in a subfolder.... e.g. `domain.com/hauk`

* How long is the map/route available for after it has expired? Can we set it for example to be deleted 24-48hrs after share link has expired?

* If I take a 6hr flight, but share my location for 12hrs, will the map show the two countries once my location is back on when I land?

* Am I right in saying, that without the android app, there is no way for my family to share their location in hauk?

**Comment by licaon-kter** [comment]
> If the install is done manually on a vps/cpanel account, can it be installed in a subfolder....

Yes

> How long is the map/route available for after it has expired?

No longer available

> If I take a 6hr flight, but share my location for 12hrs, will the map show the two countries once my location is back on when I land?

The browser shows the path from when you open it until you close it, there is no past.
Does GPS work ok in planes?

> Am I right in saying, that without the android app, there is no way for my family to share their location in hauk?

Make apps for all platforms? :)

**Comment by aljawaid** [comment]
Thank you for that fast response!

About the planes, what I meant was the gps will be lost on a plane but if the share is still active, it should stay in the same position until i connect in another country right? and then it would be drawn on the map after connection.... all if the share is still valid and not expired.

About the expiry, it would be nice if the route was still available for a set period of time that could be changed per share.

One more thing, there should be a 'distance to me' showing if the follwer has their location services on.

I'm excited to see how this project develops but at the moment i think the use case is very small and too temporary for practicality.

**Comment by bilde2910** [comment]
>If the install is done manually on a vps/cpanel account, can it be installed in a subfolder.... e.g. domain.com/hauk

Yes, this is supported. In this case, you set the [`public_url`](https://github.com/bilde2910/Hauk/blob/master/backend-php/include/config-sample.php#L246) to the URL to the directory (e.g. `https://domain.com/hauk/`. The server URL you enter in the app is then also `https://domain.com/hauk/`.

>How long is the map/route available for after it has expired? Can we set it for example to be deleted 24-48hrs after share link has expired?

The route is deleted immediately when the share expires or is stopped, and points are also pruned as you go to respect the [`max_cached_pts`](https://github.com/bilde2910/Hauk/blob/master/backend-php/include/config-sample.php#L228) setting (though this can be changed to a higher value, to store more points before they're deleted).

>If I take a 6hr flight, but share my location for 12hrs, will the map show the two countries once my location is back on when I land?

The server only keeps the `max_cached_pts` (default=3) number of points in memory at any time. The map itself will keep up to `max_shown_pts` (default=100) points visible as part of the trail behind the current position map marker, however. That said, as soon as you reload the page, the browser will only know the 3 latest points from the server, and as updates from the server arrives, it will keep adding mote points to the trail on the map until it reaches 100, at which point it starts trimming the tail.

If you don't post any location updates to Hauk while in the air, then it's possible that there could be a long leg between countries A and B visible on the map as soon as you post an update from B. If you wish to test this behavior, you can download a [GPS emulator](https://play.google.com/store/apps/details?id=com.rosteam.gpsemulator) from the Play Store to trick your phone into moving large distances.

>Am I right in saying, that without the android app, there is no way for my family to share their location in hauk?

Correct. Only an Android app is available for sharing locations, and without this, you can only view the map. An iOS app is not planned due to the cost of an Apple membership (#43) and other platforms are not supported because I don't have devices (Purism, Blackberry, Windows Phone etc.) to test with. A stable web client is also not possible currently because the technology required to maintain a sharing session in the background is still a working draft and no browsers support it yet (w3c/geolocation-sensor#30).

**Comment by bilde2910** [comment]
>About the planes, what I meant was the gps will be lost on a plane but if the share is still active, it should stay in the same position until i connect in another country right? and then it would be drawn on the map after connection.... all if the share is still valid and not expired.

This is correct.

>About the expiry, it would be nice if the route was still available for a set period of time that could be changed per share.

You could create a new issue for this, it should be possible to implement this with the current Memcached/Redis backend.

>One more thing, there should be a 'distance to me' showing if the follwer has their location services on.

There's no space for this on user markers currently, but maybe this could be possible to add in the menu suggestion in #101.

>I'm excited to see how this project develops but at the moment i think the use case is very small and too temporary for practicality.

It's intended to be a pretty narrow scope. The functionality mimics that of Glympse. For more permanent tracking, there are already many other solutions like PhoneTrack (Nextcloud addon), OwnTracks, etc.
